static/app/views/explore/hooks/useProgressiveQuery.tsx

```diff
@@ -0,0 +1,97 @@
+import useOrganization from 'sentry/utils/useOrganization';
+
+export const SAMPLING_MODE = {
+  PREFLIGHT: 'PREFLIGHT',
+  BEST_EFFORT: 'BEST_EFFORT',
+} as const;
+
+export const QUERY_MODE = {
+  SERIAL: 'serial',
+  PARALLEL: 'parallel',
+} as const;
```

Comment:
Should these be enums?

Reply:
These are enum-y, apparently enums aren't great to use in TypeScript so I've just gone ahead and used this pattern